### PR TITLE
Indexes followup

### PR DESCRIPTION
**Pull request by jjcarstens**
Based on the previous PR review. Actually use the new tag function and do some safe creates to allow us (or anyone else) to manually create indexes as needed

### CODE REVIEW

File: lib/nerves_hub/devices.ex [hunk 1]
```diff
@@ -176,7 +176,7 @@ defmodule NervesHub.Devices do
                 where(
                   query,
                   [d],
-                  fragment("array_to_string(?, ' ', ' ') ILIKE ?", d.tags, ^"%#{tag}%")
+                  fragment("string_array_to_string(?, ' ', ' ') ILIKE ?", d.tags, ^"%#{tag}%")
```
Comment (fhunleth) [inline lib/nerves_hub/devices.ex:179]: What issue does this fix?

Comment (oestrich) [reply lib/nerves_hub/devices.ex:179]: This is related to @esvinson's comment in the previous indices PR that you should use the same function as the index to trigger using the index

Comment (jjcarstens) [reply lib/nerves_hub/devices.ex:179]: Re: [this comment](https://github.com/nerves-hub/nerves_hub_web/pull/1504#discussion_r1752294710) - We created the index, but didn't end up using it. I think it was just missed in the original PR

Comment (fhunleth) [reply lib/nerves_hub/devices.ex:179]: Thanks

Comment (joshk) [reply lib/nerves_hub/devices.ex:179]: I can confirm that this change is needed, in the sense that the index isn't being used unless we use the new function.

**Pull request merged**